src/encoder.rs

```diff
@@ -112,6 +112,8 @@ pub struct Writer<W: Write> {
     info: Info,
 }
 
+const DEFAULT_BUFFER_LENGTH: usize = 4 * 1024;
```

Comment:
NB: stdlib [recommends](https://github.com/rust-lang/rust/blob/1.42.0/src/libstd/sys_common/io.rs#L1) 8k

Reply:
Moving that to its own issue #

Reply:
It's an interesting point though, I wonder where the constant came from.